store/etcd/processing.go

```diff
@@ -0,0 +1,68 @@
+package etcdstore
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	etcdclient "github.com/coreos/etcd/client"
+	"github.com/projecteru2/core/types"
+	log "github.com/sirupsen/logrus"
+)
+
+// SaveProcessing save processing status in etcd
+func (k *Krypton) SaveProcessing(ctx context.Context, opts *types.DeployOptions, nodeInfo types.NodeInfo) error {
+	///Appname/Entrypoint/Nodename/Ident
+	processingKey := filepath.Join(containerProcessingPrefix, opts.Name, opts.Entrypoint.Name, nodeInfo.Name, opts.ProcessIdent)
+	_, err := k.etcd.Create(ctx, processingKey, fmt.Sprintf("%d", nodeInfo.Deploy))
+	return err
+}
+
+// UpdateProcessing update processing status in etcd
+func (k *Krypton) UpdateProcessing(ctx context.Context, opts *types.DeployOptions, nodename string, count int) error {
+	///Appname/Entrypoint/Nodename/Ident
+	processingKey := filepath.Join(containerProcessingPrefix, opts.Name, opts.Entrypoint.Name, nodename, opts.ProcessIdent)
+	_, err := k.etcd.Update(ctx, processingKey, fmt.Sprintf("%d", count))
+	return err
+}
+
+// DeleteProcessing delete processing status in etcd
+func (k *Krypton) DeleteProcessing(ctx context.Context, opts *types.DeployOptions, nodeInfo types.NodeInfo) error {
+	///Appname/Entrypoint/Nodename/Ident
+	processingKey := filepath.Join(containerProcessingPrefix, opts.Name, opts.Entrypoint.Name, nodeInfo.Name, opts.ProcessIdent)
+	_, err := k.etcd.Delete(ctx, processingKey, &etcdclient.DeleteOptions{Recursive: true})
+	return err
+}
+
+func (k *Krypton) loadProcessing(ctx context.Context, opts *types.DeployOptions, nodesInfo []types.NodeInfo) ([]types.NodeInfo, error) {
+	processingKey := filepath.Join(containerProcessingPrefix, opts.Name, opts.Entrypoint.Name)
+	resp, err := k.etcd.Get(ctx, processingKey, &etcdclient.GetOptions{Recursive: true})
+	if err != nil {
+		if etcdclient.IsKeyNotFound(err) {
+			err = nil
+		}
+		return nodesInfo, err
```

Comment:
When `KeyNotFound`, is it really OK to return `nil` ? I'm not clear about this function.